labelbox/client.py

```diff
@@ -1211,3 +1212,27 @@ def _format_failed_rows(rows: List[str],
                     "Timed out waiting for get_data_rows_for_global_keys job to complete."
                 )
             time.sleep(sleep_time)
+
+    def get_slice(self, slice_id) -> CatalogSlice:
+        """
+        Fetches a Slice by ID.
+
+        Args:
+             slice_id (str): The ID of the Slice
+        Returns:
+            Slice
+        """
+        query_str = """
+            query getSavedQueryPyApi($id: ID!) {
+                getSavedQuery(id: $id) {
+                    id
+                    name
+                    description
+                    filter
+                    createdAt
+                    updatedAt
+                }
+            }
+        """
+        res = self.execute(query_str, {'id': slice_id})
+        return Entity.CatalogSlice(self, res['getSavedQuery'])
```

Comment:
We should think about when ModelSlice is introduced, how will this method distinguish between a CatalogSlice and ModelSlice. The method needs to be smart enough to figure that out, and create the corresponding Entity. Does SavedQuery have a "type" that we can return from graphql here?

Reply:
Yeah, there will be a `type` column in the corresponding `SavedQuery` table with possible values of `catalog` and `model`. Based on that, we could return an instance of the proper class.